omp/test/solver/idr_kernels.cpp

```diff
@@ -0,0 +1,421 @@
+    std::unique_ptr<Mtx> r;
+    std::unique_ptr<Mtx> m;
+    std::unique_ptr<Mtx> f;
+    std::unique_ptr<Mtx> g;
+    std::unique_ptr<Mtx> u;
+    std::unique_ptr<Mtx> c;
+    std::unique_ptr<Mtx> v;
+    std::unique_ptr<Mtx> p;
+    std::unique_ptr<Mtx> omega;
+    std::unique_ptr<Mtx> tht;
+    std::unique_ptr<Mtx> residual_norm;
+    std::unique_ptr<gko::Array<gko::stopping_status>> stop_status;
+
+    std::unique_ptr<Mtx> d_x;
+    std::unique_ptr<Mtx> d_b;
+    std::unique_ptr<Mtx> d_r;
+    std::unique_ptr<Mtx> d_m;
+    std::unique_ptr<Mtx> d_f;
+    std::unique_ptr<Mtx> d_g;
+    std::unique_ptr<Mtx> d_u;
+    std::unique_ptr<Mtx> d_c;
+    std::unique_ptr<Mtx> d_v;
+    std::unique_ptr<Mtx> d_p;
+    std::unique_ptr<Mtx> d_omega;
+    std::unique_ptr<Mtx> d_tht;
+    std::unique_ptr<Mtx> d_residual_norm;
+    std::unique_ptr<gko::Array<gko::stopping_status>> d_stop_status;
+};
+
+
+TEST_F(Idr, IdrInitializeIsEquivalentToRef)
+{
+    initialize_data();
+
+    gko::kernels::reference::idr::initialize(ref, m.get(), p.get(), true,
+                                             stop_status.get());
+    gko::kernels::omp::idr::initialize(omp, d_m.get(), d_p.get(), true,
+                                       d_stop_status.get());
+
+    GKO_ASSERT_MTX_NEAR(m, d_m, 1e-14);
+    GKO_ASSERT_MTX_NEAR(p, d_p, 1e-14);
+}
+
+
+TEST_F(Idr, IdrStep1IsEquivalentToRef)
+{
+    initialize_data();
+
+    gko::size_type k = 2;
+    gko::kernels::reference::idr::step_1(ref, k, m.get(), f.get(), r.get(),
+                                         g.get(), c.get(), v.get(),
+                                         stop_status.get());
+    gko::kernels::omp::idr::step_1(omp, k, d_m.get(), d_f.get(), d_r.get(),
+                                   d_g.get(), d_c.get(), d_v.get(),
+                                   d_stop_status.get());
+
+    GKO_ASSERT_MTX_NEAR(c, d_c, 1e-14);
+    GKO_ASSERT_MTX_NEAR(v, d_v, 1e-14);
+}
+
+
+TEST_F(Idr, IdrStep2IsEquivalentToRef)
+{
+    initialize_data();
+
+    gko::size_type k = 2;
+    gko::kernels::reference::idr::step_2(ref, k, omega.get(), v.get(), c.get(),
+                                         u.get(), stop_status.get());
+    gko::kernels::omp::idr::step_2(omp, k, d_omega.get(), d_v.get(), d_c.get(),
+                                   d_u.get(), d_stop_status.get());
+
+    GKO_ASSERT_MTX_NEAR(u, d_u, 1e-14);
+}
+
+
+TEST_F(Idr, IdrStep3IsEquivalentToRef)
+{
+    initialize_data();
+
+    gko::size_type k = 2;
+    gko::kernels::reference::idr::step_3(ref, k, p.get(), g.get(), v.get(),
+                                         u.get(), m.get(), f.get(), omega.get(),
+                                         r.get(), x.get(), stop_status.get());
+    gko::kernels::omp::idr::step_3(
+        omp, k, d_p.get(), d_g.get(), d_v.get(), d_u.get(), d_m.get(),
+        d_f.get(), d_omega.get(), d_r.get(), d_x.get(), d_stop_status.get());
+
+    GKO_ASSERT_MTX_NEAR(g, d_g, 1e-14);
+    GKO_ASSERT_MTX_NEAR(u, d_u, 1e-14);
```

Comment:
Looks like `v` (`g_k` in the parameter list) is also computed on, so probably it should also be checked.
```suggestion
    GKO_ASSERT_MTX_NEAR(g, d_g, 1e-14);
    GKO_ASSERT_MTX_NEAR(v, d_v, 1e-14);
    GKO_ASSERT_MTX_NEAR(u, d_u, 1e-14);
```